### DISCUSSION TITLE
planetscale prodcution and development branch string is the same

### DISCUSSION THREAD

**Discussion by isweir** [planetscale/discussion#473] [Bug Reports] [answered]
Hi, I'm new to planetscale, i can't tell which one is production and which one is development db branch due to that their username string and pass is the same for production and development branch, so when i try npx prisma db push on node.js, I got following error:

Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": MySQL database "twitter-clone" at "aws.connect.psdb.cloud"
Error: direct DDL is disabled
   0: sql_schema_connector::apply_migration::migration_step
           with step=DropTable { table_id: TableId(1) }
             at schema-engine\connectors\sql-schema-connector\src\apply_migration.rs:21 
   1: sql_schema_connector::apply_migration::apply_migration
             at schema-engine\connectors\sql-schema-connector\src\apply_migration.rs:10 
   2: schema_core::state::SchemaPush
             at schema-engine\core\src\state.rs:433
             
PS: This is frustrated.

**Comment by isweir** [comment]
More dig on the problem:  when I click on the main branch and click on connect button, the form show up always indicate i'm in dev branch and role is admin, the main branch is no where to be found, so the db string always show dev branch db string i guess. so where is main branch string?

**Reply by isweir** [reply]
![Screenshot 2023-05-23 171803](https://github.com/planetscale/discussion/assets/118220037/08fab2ad-ae38-4ea1-bd75-108e53edce0d)
![Screenshot 2023-05-23 171942](https://github.com/planetscale/discussion/assets/118220037/24d6bc79-bbb6-473e-86a1-2fc7e69f20bd)

**Reply by isweir** [reply]
![Screenshot 2023-05-23 172049](https://github.com/planetscale/discussion/assets/118220037/cf9ce9b1-c5f0-4f7c-a15d-afd32045c81e)

**Comment by isweir** [comment]
![Screenshot 2023-05-23 172213](https://github.com/planetscale/discussion/assets/118220037/cb3ccf62-d0b5-43e1-aef8-1c9fe9c9f768)

**Comment by isweir** [comment]
i got ticket assisted that I demoted my dev branch from "this is a production branch" and it worked, no errors anymore.

**Reply by isweir** [reply]
how ever, this might be stupid, i still didn't know is prod and dev branch share the same db string?

**Reply by jgreet** [reply, marked as answer]
No, when you create a new branch you also to have to create new credentials for it.

![image](https://github.com/planetscale/discussion/assets/391331/f67ccd2f-7400-4e30-bfdb-d2cdec2c8ce5)

 - In this dialog click **New password**
- In the next form, pick your dev branch from the **Branch** dropdown
- Click **Create password** and you'll see the new credentials for your dev branch

**Reply by isweir** [reply]
thanks problem solved.
